packages/bot-web-ui/src/components/dashboard/bot-builder/toolbox/__tests__/search-icon.spec.tsx

```diff
@@ -19,18 +22,39 @@ const mocked_props = {
     is_search_loading: false,
     onClick: jest.fn(),
 };
+jest.mock('@deriv/bot-skeleton/src/scratch/blockly', () => jest.fn());
+jest.mock('@deriv/bot-skeleton/src/scratch/dbot', () => ({
+    saveRecentWorkspace: jest.fn(),
+    unHighlightAllBlocks: jest.fn(),
+}));
+jest.mock('@deriv/bot-skeleton/src/scratch/hooks/block_svg', () => jest.fn());
 
 describe('SearchIcon', () => {
     it('should render the SearchIcon component', () => {
-        render(<SearchIcon {...mocked_props} />);
-
+        const mock_store = mockStore({});
+        const mock_DBot_store = mockDBotStore(mock_store, mock_ws);
+        render(
+            <StoreProvider store={mock_store}>
+                <DBotStoreProvider ws={mock_ws} mock={mock_DBot_store}>
+                    <SearchIcon {...mocked_props} />
+                </DBotStoreProvider>
+            </StoreProvider>
+        );
```

Comment:
Can we use before each for this? As it has been used multiple times in this test file?